compaction/compaction_manager.hh

```diff
@@ -60,6 +60,9 @@ public:
     };
 private:
     struct compaction_state {
+        // FIXME: remove once we complete the transition to table_state.
```

Comment:
We'll probably need a backpointer to the table, either from the table_state or from here when we have multiple table states / compaction groups per table.

Reply:
I already did some initial work on compaction groups, and realized compaction_group class, which is stored in replica::table may need a reference to the underlying table. compaction_manager will never reference replica::table anymore from the moment this series is merged.

Reply:
replica::table's subclass to compaction::compaction_group can have a backpointer. compaction::compaction_group should not know about tables.